### DISCUSSION TITLE
Help with Capytaine & GBM for OWC

### DISCUSSION THREAD

**Discussion by samir-miloud** [capytaine/capytaine#582] [Q&A]
Hi,

I’m currently working on modeling the Sparbuoy OWC case using Capytaine, employing the Generalized Body Modes (GBM) method. I need to incorporate 7 degrees of freedom (DoFs) into the simulation. However, Capytaine only provides the 6 standard DoFs (surge, sway, heave, roll, pitch, yaw), and I am missing an additional DoF for the water column height.
Does anyone have experience with adding a 7th DoF in Capytaine, or any suggestions on how to handle this with the GBM method?

Thanks!

**Comment by mancellin** [comment]
Hi,

Capytaine can handle arbitrary dofs (there is [an example in the documentation](https://capytaine.github.io/v2.2/user_manual/body.html#dofs)), which include the generalized body modes used for OWC.

I've never done this myself. Testing and documenting an example of OWC with Capytaine is among the most requested feature, but I won't have time and funding to work on that in the next few months.

**Comment by samir-miloud** [comment]
Hi @mancellin,

Thank you very much for your response and the information about Capytaine. I reviewed the example in the documentation as you suggested, which was helpful in understanding the available options.
However, I am facing some specific challenges in my Sparbuoy OWC modeling project. I use Capytaine for hydrodynamic calculations to obtain the added mass matrices, hydrodynamic damping, hydrostatic stiffness, and excitation forces. Then, I use WEC-Sim for dynamic calculations.
In the context of the Sparbuoy study, 6 degrees of freedom (DoFs) are used for standard movements such as heave, pitch, roll, etc. My challenge is to integrate a 7th DoF specifically dedicated to the water column height, in addition to the 6 standard DoFs provided by Capytaine.
I find it particularly difficult to integrate this 7th DoF into my simulations. Any assistance or suggestions on how to handle this issue or integrate the 7th DoF would be greatly appreciated.

Thank's

Best regards,
Samir

**Reply by mancellin** [reply]
Capytaine will get you 7x7 matrices for added mass and radiation damping, as well as the excitations force vector. You can use them to solve for the dynamics of your device in frequency domain (RAO).
WEC-SIM can be used to simulate devices in time domain, but I'm not sure whether WEC-SIM supports non-rigid-body dofs like your 7th dof.
